Error while updating Moya with carthge

<!--
Please let us know what version of Moya you are using, so we can better pinpoint and/or solve your issue.

Please wrap code blocks in backticks, like so:

```swift
*your code goes here*
```

The code will automatically get its syntax highlighted, and doesn't need to be indented 4 spaces to be shown as code.

When referencing a dependency manager-related issue (think CocoaPods, Carthage, SwiftPM), please add its configuration file and version to the issue.
It would be helpful to put the contents in a code block too, using ```ruby for CocoaPods and ```swift for SwiftPM.

Also please make sure your title describes your problem well. Questions end with a question mark.
-->
Hello,
I'm using carthage version 0.26.2, and when trying to update with `carthage update --platform iOS` I get, only for Moya, this error:
```
*** Skipped installing Moya.framework binary due to the error:
	"Incompatible Swift version - framework was built with 4.0 (swiftlang-900.0.65.2 clang-900.0.37) and the local version is 4.0.2 (swiftlang-900.0.69.2 clang-900.0.38)."
*** Checking out Moya at "10.0.0" 
```

Here is my Cartfile:
```github "Moya/Moya" "10.0.0"```

Would you have any idea of why or how I could make this work ?

I'm seeing a similar issue with the now latest swift `Apple Swift version 4.0.3 (swiftlang-900.0.74.1 clang-900.0.39.2)` but the log shows a different error

```
Carthage/Checkouts/Moya/Sources/ReactiveMoya/MoyaProvider+Reactive.swift:36:13: error: value of type 'CompositeDisposable' has no member 'observeEnded'
            lifetime.observeEnded {
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -29,6 +29,9 @@ jobs:
           name: Update Homebrew
           command: brew update
       - run:
+          name: Test Carthage Build before installing SwiftLint
+          command: carthage build --no-skip-current --cache-builds
+      - run:
           name: Install Swiftlint
           command: brew install swiftlint
       - run:

--- a/Changelog.md
+++ b/Changelog.md
@@ -1,4 +1,6 @@
 # Next
+### Fixed
+- Fixed Carthage compatibility by disabling the SwiftLint build phase in release builds. [#1619](https://github.com/Moya/Moya/pull/1619) by [@Dschee](https://github.com/Dschee).
 
 # [11.0.1] - 2018-02-26
 

--- a/Examples/Basic/ViewController.swift
+++ b/Examples/Basic/ViewController.swift
@@ -17,8 +17,8 @@ class ViewController: UITableViewController {
 
     fileprivate func showAlert(_ title: String, message: String) {
         let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
-        let ok = UIAlertAction(title: "OK", style: .default, handler: nil)
-        alertController.addAction(ok)
+        let okAction = UIAlertAction(title: "OK", style: .default, handler: nil)
+        alertController.addAction(okAction)
         present(alertController, animated: true, completion: nil)
     }
 
@@ -109,12 +109,12 @@ class ViewController: UITableViewController {
         var usernameTextField: UITextField?
 
         let promptController = UIAlertController(title: "Username", message: nil, preferredStyle: .alert)
-        let ok = UIAlertAction(title: "OK", style: .default) { _ in
+        let okAction = UIAlertAction(title: "OK", style: .default) { _ in
             if let username = usernameTextField?.text {
                 self.downloadRepositories(username)
             }
         }
-        promptController.addAction(ok)
+        promptController.addAction(okAction)
         promptController.addTextField { textField in
             usernameTextField = textField
         }

--- a/Examples/Multi-Target/ViewController.swift
+++ b/Examples/Multi-Target/ViewController.swift
@@ -19,8 +19,8 @@ class ViewController: UITableViewController {
 
     fileprivate func showAlert(_ title: String, message: String) {
         let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
-        let ok = UIAlertAction(title: "OK", style: .default, handler: nil)
-        alertController.addAction(ok)
+        let okAction = UIAlertAction(title: "OK", style: .default, handler: nil)
+        alertController.addAction(okAction)
         present(alertController, animated: true, completion: nil)
     }
 
@@ -112,12 +112,12 @@ class ViewController: UITableViewController {
         var usernameTextField: UITextField?
 
         let promptController = UIAlertController(title: "Username", message: nil, preferredStyle: .alert)
-        let ok = UIAlertAction(title: "OK", style: .default) { _ in
+        let okAction = UIAlertAction(title: "OK", style: .default) { _ in
             if let username = usernameTextField?.text {
                 self.downloadRepositories(username)
             }
         }
-        promptController.addAction(ok)
+        promptController.addAction(okAction)
         promptController.addTextField { textField in
             usernameTextField = textField
         }

--- a/Moya.xcodeproj/project.pbxproj
+++ b/Moya.xcodeproj/project.pbxproj
@@ -721,7 +721,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\nfi";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    if which swiftlint > /dev/null; then\n        swiftlint\n    else\n        echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\n    fi\nfi";
 		};
 		46AE31F31F86DDFC004E4236 /* Swiftlint */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -735,7 +735,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\nfi";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    if which swiftlint > /dev/null; then\n        swiftlint\n    else\n        echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\n    fi\nfi";
 		};
 		46AE31F41F86DED0004E4236 /* Swiftlint */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -749,7 +749,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\nfi";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    if which swiftlint > /dev/null; then\n        swiftlint\n    else\n        echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\n    fi\nfi";
 		};
 		46AE31F51F86DEE2004E4236 /* Swiftlint */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -763,7 +763,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\nfi";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    if which swiftlint > /dev/null; then\n        swiftlint\n    else\n        echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\n    fi\nfi";
 		};
 		46AE31F61F86DEEF004E4236 /* Swiftlint */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -777,7 +777,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\nfi";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    if which swiftlint > /dev/null; then\n        swiftlint\n    else\n        echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\n    fi\nfi";
 		};
 		46AE31F71F86DEFE004E4236 /* Swiftlint */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -791,7 +791,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint\nelse\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\nfi";
+			shellScript = "if [ \"${CONFIGURATION}\" = \"Debug\" ]; then\n    if which swiftlint > /dev/null; then\n        swiftlint\n    else\n        echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint to lint!\"\n    fi\nfi";
 		};
 		B33B10BA6802FCF699147E8B /* Copy Carthage Frameworks */ = {
 			isa = PBXShellScriptBuildPhase;

--- a/Sources/Moya/MultiTarget.swift
+++ b/Sources/Moya/MultiTarget.swift
@@ -48,7 +48,7 @@ public enum MultiTarget: TargetType {
     /// The embedded `TargetType`.
     public var target: TargetType {
         switch self {
-        case .target(let t): return t
+        case .target(let target): return target
         }
     }
 }

--- a/Sources/Moya/ValidationType.swift
+++ b/Sources/Moya/ValidationType.swift
@@ -38,8 +38,8 @@ extension ValidationType: Equatable {
              (.successCodes, .successCodes),
              (.successAndRedirectCodes, .successAndRedirectCodes):
             return true
-        case (.customCodes(let c1), .customCodes(let c2)):
-            return c1 == c2
+        case (.customCodes(let code1), .customCodes(let code2)):
+            return code1 == code2
         default:
             return false
         }

--- a/Tests/ErrorTests.swift
+++ b/Tests/ErrorTests.swift
@@ -82,8 +82,8 @@ final class ErrorTests: QuickSpec {
                 switch result {
                 case let .failure(error):
                     switch error {
-                    case let .underlying(e, _):
-                        expect(e as NSError) == underlyingError
+                    case let .underlying(error, _):
+                        expect(error as NSError) == underlyingError
                     default:
                         XCTFail("expected to get underlying error")
                     }

--- a/Tests/NetworkLoggerPluginSpec.swift
+++ b/Tests/NetworkLoggerPluginSpec.swift
@@ -127,11 +127,11 @@ final class NetworkLoggerPluginSpec: QuickSpec {
 
 private class TestStreamRequest: RequestType {
     var request: URLRequest? {
-        var r = URLRequest(url: URL(string: url(GitHub.zen))!)
-        r.allHTTPHeaderFields = ["Content-Type": "application/json"]
-        r.httpBodyStream = InputStream(data: "cool body".data(using: .utf8)!)
+        var request = URLRequest(url: URL(string: url(GitHub.zen))!)
+        request.allHTTPHeaderFields = ["Content-Type": "application/json"]
+        request.httpBodyStream = InputStream(data: "cool body".data(using: .utf8)!)
 
-        return r
+        return request
     }
 
     func authenticate(user: String, password: String, persistence: URLCredential.Persistence) -> Self {
@@ -145,11 +145,11 @@ private class TestStreamRequest: RequestType {
 
 private class TestBodyRequest: RequestType {
     var request: URLRequest? {
-        var r = URLRequest(url: URL(string: url(GitHub.zen))!)
-        r.allHTTPHeaderFields = ["Content-Type": "application/json"]
-        r.httpBody = "cool body".data(using: .utf8)
+        var request = URLRequest(url: URL(string: url(GitHub.zen))!)
+        request.allHTTPHeaderFields = ["Content-Type": "application/json"]
+        request.httpBody = "cool body".data(using: .utf8)
 
-        return r
+        return request
     }
 
     func authenticate(user: String, password: String, persistence: URLCredential.Persistence) -> Self {
@@ -163,11 +163,11 @@ private class TestBodyRequest: RequestType {
 
 private class TestCurlBodyRequest: RequestType, CustomDebugStringConvertible {
     var request: URLRequest? {
-        var r = URLRequest(url: URL(string: url(GitHub.zen))!)
-        r.allHTTPHeaderFields = ["Content-Type": "application/json"]
-        r.httpBody = "cool body".data(using: .utf8)
+        var request = URLRequest(url: URL(string: url(GitHub.zen))!)
+        request.allHTTPHeaderFields = ["Content-Type": "application/json"]
+        request.httpBody = "cool body".data(using: .utf8)
 
-        return r
+        return request
     }
 
     func authenticate(user: String, password: String, persistence: URLCredential.Persistence) -> Self {

--- a/Tests/SignalProducer+MoyaSpec.swift
+++ b/Tests/SignalProducer+MoyaSpec.swift
@@ -169,8 +169,8 @@ final class SignalProducerMoyaSpec: QuickSpec {
 
                 var receivedJSON: [String: String]?
                 signal.mapJSON().startWithResult { result in
-                    if case .success(let _json) = result,
-                        let json = _json as? [String: String] {
+                    if case .success(let response) = result,
+                        let json = response as? [String: String] {
                         receivedJSON = json
                     }
                 }

